src/ViewLibrarian.java

```diff
@@ -0,0 +1,77 @@
+import java.awt.BorderLayout;
+import java.awt.EventQueue;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.border.EmptyBorder;
+import javax.swing.JTable;
+
+public class ViewLibrarian extends JFrame {
+
+	private JPanel contentPane;
+	private JTable table;
+
+	public static void main(String[] args) {
+		EventQueue.invokeLater(new Runnable() {
+			public void run() {
+				try {
+					ViewLibrarian frame = new ViewLibrarian();
+					frame.setVisible(true);
+				} catch (Exception e) {
+					e.printStackTrace();
+				}
+			}
+		});
+	}
+
+	/**
+	 * Create the frame.
+	 */
+	public ViewLibrarian() {
+		setDefaultCloseOperation(JFrame.HIDE_ON_CLOSE);
+		setBounds(100, 100, 450, 300);
+		contentPane = new JPanel();
+		contentPane.setBorder(new EmptyBorder(5, 5, 5, 5));
+		contentPane.setLayout(new BorderLayout(0, 0));
+		setContentPane(contentPane);
+		String data[][]=null;
+		String column[]=null;
+		try{
+			Connection con=DB.getConnection();
+			PreparedStatement ps=con.prepareStatement("select * from librarian",ResultSet.TYPE_SCROLL_SENSITIVE,ResultSet.CONCUR_UPDATABLE);
+			ResultSet rs=ps.executeQuery();

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *rs*. The resource is referenced in statements at the following lines: 46, 47, 49, 56, 57, 58, 62, 64. The resource closure statement is at line: 68. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by ResultSet.getString. Either a) close *rs* in a try-finally block or b) close the resource by declaring *rs* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).